codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentBuilderGenerator.kt

```diff
@@ -118,6 +128,16 @@ class FluentBuilderGenerator(
         writable {
             docs("Fluent builder constructing a request to `${operationType.name}`.\n")
             documentShape(operation, model, autoSuppressMissingDocs = false)
+            if (codegenContext.protocolImpl?.httpBindingResolver?.handlesEventStreamInitialResponse(operation) == true) {
+                docs(
+                    """
+                    [`${outputType.name}`]($outputType) contains an event stream field as well as one or more non-event stream fields.
+                    Due to its current implementation, the non-event stream fields are not fully deserialized
+                    until the [`send`](Self::send) method completes. As a result, accessing these fields of the operation
+                    output struct within an interceptor may return uninitialized values.
+                    """,
```

Comment:
This warning message has me thinking about the design. While the current implementation is workable for this PR, I have a few minor questions:

1. Does this partial deserialization issue affect only interceptors, or could it also impact user-written code?
2. Could we consider an alternative design in future where we use an `enum` to represent streaming messages? For example:
. One variant for the initial message (containing only fields that can be part of the initial message)
. Another variant for the complete message